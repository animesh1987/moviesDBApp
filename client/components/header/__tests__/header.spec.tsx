import * as React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import { Header } from '../index';

describe('Header Test', () => {
  const movie = {
    title: 'Test',
    overview: 'Test overview',
    release_date: "20-04-04",
    runtime: 144,

  };

  const genre = 'title';

  const onChangeInput = jest.fn();
  const reset = jest.fn();
  const triggerSearch = jest.fn();
  const changeSearchInput = jest.fn();
  const getSearchInput = jest.fn();
  const toggleType = jest.fn();

  it('Header without movie renders correctly', () => {
    const isMovieSelected = false;
    const component = create(
      <Header
        reset={reset}
        onSearchClick={triggerSearch}
        onChangeInput={changeSearchInput}
        searchInputEnter={getSearchInput}
        toggleType={toggleType}
        searchInput=""
        movie={movie}
        genreSelected={genre}
        isMovieSelected={isMovieSelected} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Header with movie renders correctly', () => {
    const isMovieSelected = true;
    const component = create(
      <Header
        reset={reset}
        onSearchClick={triggerSearch}
        onChangeInput={changeSearchInput}
        searchInputEnter={getSearchInput}
        toggleType={toggleType}
        searchInput=""
        movie={movie}
        genreSelected={genre}
        isMovieSelected={isMovieSelected} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('active added successfully', () => {
    const isMovieSelected = false;
    const component = shallow(
      <Header
        reset={reset}
        onSearchClick={triggerSearch}
        onChangeInput={changeSearchInput}
        searchInputEnter={getSearchInput}
        toggleType={toggleType}
        searchInput=""
        movie={movie}
        genreSelected={genre}
        isMovieSelected={isMovieSelected} />
    );
    expect(component.find('.header-bar')).to.have.length(1);//.text());//.simulate('click');
    expect(3).toBe(3);
  });
});