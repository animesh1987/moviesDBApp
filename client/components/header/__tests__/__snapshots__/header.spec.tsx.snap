// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test Header with movie renders correctly 1`] = `
<div
  className="header-bar"
>
  <div
    className="layout-row layout-align-space-between movie-selected"
  >
    <h3>
      netflixroulette
    </h3>
    <button
      className=""
      onClick={[Function]}
    >
      Search
    </button>
  </div>
  <div
    className="movie-detail layout-row"
  >
    <img
      alt="Movie Banner"
      src={undefined}
    />
    <div
      className="layout-column"
    >
      <div
        className="layout-row layout-align-center"
      >
        <span
          className="title"
        >
          Test
        </span>
        <span
          className="average"
        />
      </div>
      <div
        className="tagline"
      />
      <div
        className="layout-row"
      >
        <span
          className="date"
        >
          20-0
        </span>
        <span
          className="length"
        >
          144
           min
        </span>
      </div>
      <p
        className="description"
      >
        Test overview
      </p>
    </div>
  </div>
</div>
`;

exports[`Header Test Header without movie renders correctly 1`] = `
<div
  className="header-bar"
>
  <h3>
    netflixroulette
  </h3>
  <div
    className="search-title"
  >
    Find your movie
  </div>
  <div
    className="search-input-container"
  >
    <input
      onChange={undefined}
      onKeyPress={undefined}
      placeholder="Titles, people, genres..."
      type="text"
      value=""
    />
    <i
      className="material-icons"
    >
      keyboard_return
    </i>
  </div>
  <div
    className="layout-row layout-align-space-between"
  >
    <div
      className="header-bar__searchBy"
    >
      <span
        className="search-by"
      >
        Search By
      </span>
      <button
        className="active"
        onClick={[Function]}
      >
        Title
      </button>
      <button
        className=""
        onClick={[Function]}
      >
        Genre
      </button>
    </div>
    <button
      className="primary long"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;
